pkg/controller/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -855,22 +855,37 @@ func (r *ReconcileAWSPrivateLink) reconcileVPCEndpoint(awsClient *awsClient,
 	endpointLog := logger.WithField("tag:key", aws.StringValue(tag.Name)).WithField("tag:value", aws.StringValueSlice(tag.Values))
 
 	var vpcEndpoint *ec2.VpcEndpoint
-	resp, err := awsClient.hub.DescribeVpcEndpoints(&ec2.DescribeVpcEndpointsInput{
+
+	input := &ec2.DescribeVpcEndpointsInput{
 		Filters: []*ec2.Filter{tag},
-	})
-	if err != nil {
-		endpointLog.WithError(err).Error("error getting VPC Endpoint")
-		return modified, nil, err
 	}
-	if len(resp.VpcEndpoints) == 0 {
+
+	var allVpcEndpoints []*ec2.VpcEndpoint
+	for {
+		resp, err := awsClient.hub.DescribeVpcEndpoints(input)
+		if err != nil {
+			endpointLog.WithError(err).Error("error getting VPC Endpoint")
+			return modified, nil, err
+		}
+
+		allVpcEndpoints = append(allVpcEndpoints, resp.VpcEndpoints...)
```

Comment:
For the current fix, I'm not sure that we need to keep looping until we get *all* the VPCEs. We only care about one of them, so we could `break` as soon as we get a nonempty page. Otherwise we're potentially incurring a bunch of extra network calls & bytes. (Does throttling get applied to page requests??)

However, this may also be where cleanup would need to happen. Again, since we only want one VPCE for this filter, we know that we want to scrub `allVpcEndpoints[1:]`.